### PR TITLE
Add with-credentials parameter for CORS

### PR DESCRIPTION
**Pull request by wq9**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/218)

### CODE REVIEW

**Comment by YuriyVaadin** [comment]
Dear @wq9,

Thank you very much for your Pull Request! Unfortunately we are not able to merge it right away due to the tight schedule. But we will try to manage it as soon as possible.

**Comment by limonte** [comment]
@Legioth your opinion on this PR?

**Comment by Legioth** [comment]
I think this is a good feature to have in some cases, and in other cases it won't get in the way since the property is `false` by default.

Documentation and test coverage might be slightly lacking, but that can of course be fixed in a separate PR.

**Comment by web-padawan** [comment]
@limonte BTW one of those extra whitespaces is not handled by ESLint, it seems that enabling [space-in-parens](https://github.com/vaadin/eslint-config-vaadin/blob/master/index.js#L241) rule could help.

**Comment by limonte** [comment]
@web-padawan thanks for noticing that small detail! https://github.com/vaadin/eslint-config-vaadin/pull/9

File: vaadin-upload.html [hunk 1]
```diff
@@ -614,6 +622,9 @@
       if (this.timeout) {
         xhr.timeout = this.timeout;
       }
+      if (this.withCredentials ) {
```
Comment (stramel) [inline vaadin-upload.html]: Remove extra space before the close paren

File: vaadin-upload.html [hunk 2]
```diff
@@ -614,6 +622,9 @@
...
+        xhr.withCredentials  = this.withCredentials ;
```
Comment (stramel) [inline vaadin-upload.html]: Remove extra space before the equal sign and semi-colon

**Comment by manolo** [comment]
@wq9, we have just merged the conversion to pure polymer-2 element of this component. Can you rebase on master? Also do you mind to add a small test for this?  

---

Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/218#-:-KwAlP1ewBIOrzoZh_zW:b-llabx7)*

**Comment by wq9** [comment]
@manolo I merged with the conversion to pure polymer-2 element. How do I test for with-credentials?

**Comment by manolo** [comment]
This should be enough:

```
        it('should propagate with-credentials to the xhr', done => {
          upload.withCredentials = true;
          upload.addEventListener('upload-start', e => {
            e.preventDefault();
            expect(e.detail.xhr.withCredentials).to.be.true;
            done();
          });                    
          upload._uploadFile(file);
        });
```

Add it to the `test/upload.html` 

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-upload.html, line 500 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/218#-KwEXets8_FDwr7dEaUp:-KwEXets8_FDwr7dEaUq:bpob6ub) ([raw file](https://github.com/vaadin/vaadin-upload/blob/ab8f6662e66e7318a1d7cd38284db9ad09baf922/vaadin-upload.html#L500)):*
> ```HTML
>             xhr.timeout = this.timeout;
>           }
>           if (this.withCredentials) {
> ```

Not need to have the `if` block. Just te line setting the xhr property is enough since the property was defined as boolean.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/218#-:-KwEU3yaAfGPVEqthp-G:bx1s3v4)*

**Comment by wq9** [comment]
Thanks @manolo! I added the test and removed the if block.

**Comment by manolo** [comment]
:lgtm: Thanks !

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/218#-:-Kw_pEb8K1t-Z7I7J9WR:b-qyacn)*

**Pull request merged**